src/MAGMA.jl

```diff
@@ -1,5 +1,104 @@
 module MAGMA
 
-greet() = print("Hello World!")
+export device_reset,
+       MagmaVector, MagmaMatrix, get,
+       axpy!, dot,
+       gemv!
 
-end # module
+import Base: length,size,dot
+
+using  CUDArt
+import CUDArt: device_reset
+
+const magmatypes = Union{Float32, Float64, Complex64, Complex128}
+const magmatypeslist =  (Float32,Float64,Complex64,Complex128)
+const typechar = Dict(
+    Float32 => 's',
+    Float64 => 'd',
+    Complex64 => 'c',
+    Complex128 => 'z'
+)
+
+type MagmaVector{T <: magmatypes}
+    ptr::Ptr{T}
+    length::Int
+end
+
+type MagmaMatrix{T <: magmatypes}
+    ptr::Ptr{T}
+    size::Tuple{Int, Int}
+end
+
+length(x::MagmaVector) = x.length
+size(x::MagmaMatrix)   = x.size
+size(x::MagmaMatrix,i) = x.size[i]
+
+include("enums.jl")
+const libmagma = Libdl.find_library("libmagma", ["/usr/local/magma/lib"])
```

Comment:
could test this line to see if this is working on your own laptop first? the library path might not be the correct path to your own.

Let's move them line by line and make sure each line is working, so we don't mess up our own brains. It would be nice if you could add some unit test while moving them to here.